src/main/java/meteordevelopment/meteorclient/systems/modules/combat/ArrowDodge.java

```diff
@@ -79,59 +94,53 @@
         new Vec3d(-1, 0, -1)
     );
 
+    private final ProjectileEntitySimulator simulator = new ProjectileEntitySimulator();
+    private final Pool<Vec3> vec3s = new Pool<>(Vec3::new);
+    private final List<Vec3> points = new ArrayList<>();
+
     public ArrowDodge() {
         super(Categories.Combat, "arrow-dodge", "Tries to dodge arrows coming at you.");
     }
 
     @EventHandler
     private void onTick(TickEvent.Pre event) {
-        Box playerHitbox = mc.player.getBoundingBox();
-        playerHitbox = playerHitbox.expand(0.6);
-
-        double speed = moveSpeed.get();
-
-        for (Entity e : mc.world.getEntities()) {
-            if (!(e instanceof ProjectileEntity)) continue;
-            if (e instanceof PersistentProjectileEntity && ((ProjectileInGroundAccessor) e).getInGround()) continue;
-            if (ignoreOwn.get() && ((ProjectileEntityAccessor) e).getOwnerUuid().equals(mc.player.getUuid())) continue;
-
-            List<Box> futureArrowHitboxes = new ArrayList<>();
-
-            for (int i = 0; i < arrowLookahead.get(); i++) {
-                Vec3d nextPos = e.getPos().add(e.getVelocity().multiply(i / 5.0f));
-                futureArrowHitboxes.add(new Box(
-                    nextPos.subtract(e.getBoundingBox().getXLength() / 2, 0, e.getBoundingBox().getZLength() / 2),
-                    nextPos.add(e.getBoundingBox().getXLength() / 2, e.getBoundingBox().getYLength(), e.getBoundingBox().getZLength() / 2)));
+        for (Vec3 point: points) vec3s.free(point);
+        points.clear();
+        mc.world.getEntities().forEach(e -> {
+            if (!(e instanceof ProjectileEntity)) return;
+            if (ignoreOwn.get() && ((ProjectileEntityAccessor) e).getOwnerUuid().equals(mc.player.getUuid())) return;
+            if (!simulator.set(e, accurate.get(), 0.5D)) return;
+            if (simulationSteps.get() == 0) {
+                while (true) {
+                    points.add(vec3s.get().set(simulator.pos));
+                    if (simulator.tick() != null) break;
+                }
+            } else {
+                for (int i = 0; i < simulationSteps.get(); i++) {
```

Comment:
You could merge the two loops by using `Integer.MAX_VALUE` as the limit if `simulationSteps` is 0